lib/functions_framework/legacy_event_converter.rb

```diff
@@ -129,12 +130,20 @@ def construct_cloud_event context, data
                                time:              context[:timestamp]
     end
 
-    def convert_source service, resource
+    def convert_source service, resource, domain
       return ["//#{service}/#{resource}", nil] unless CE_SERVICE_TO_RESOURCE_RE.key? service
 
       match = CE_SERVICE_TO_RESOURCE_RE[service].match resource
       return [nil, nil] unless match
-      ["//#{service}/#{match[1]}", match[2]]
+
+      if service == "firebasedatabase.googleapis.com"
+        raise "Invalid firebasedatabase event: domain is nil" if domain.nil?
+        location = "us-central1"
+        location = domain.split(".")[0] if domain != "firebaseio.com"
```

Comment:
Should we instead check explicitly for `/^([\w-]+)\.firebasedatabase\.app$/` to get a location? Is there any possibility of additional domains that aren't `.firebasedatabase.app` or `firebaseio.com`, or are we sure those are the only two possible forms that will ever happen?

Reply:
I am not positive about this. Let's make it more strict for now. I will follow up with Jon Skeet to confirm whether or not other domains are possible.